File: game_of_life/mrugger/ruby/game.grid.rb

class GameGrid

  private

  @gridsize
  @grid


  public

  def initialize(gridsize)
    @gridsize = gridsize
    @grid = Array.new(@gridsize.x) { |i| Array.new(@gridsize.y) { |i| 0 } }
  end


  # Grid interface

  def get_width
    @gridsize.y
  end

  def get_height
    @gridsize.x
  end

  def node_value(x, y)
    raise RangeError, "x (#{x}) parameter out of range" if x < 0 || x >= get_height
    raise RangeError, "y (#{y}) parameter out of range" if y < 0 || y >= get_width
    @grid[x][y]
  end

  def in_range?(x, y)
    return false if x < 0 || x >= get_height || y < 0 || y >= get_width
    return true
  end


  protected

  def set_node_value(x, y, value)
    raise RangeError, "x (#{x}) parameter out of range" if x < 0 || x >= get_height
    raise RangeError, "y (#{y}) parameter out of range" if y < 0 || y >= get_width
    @grid[x][y] = value
  end

end

File: game_of_life/mrugger/java/GameGrid.java


class GameGrid implements IGridable
{
  private Pair _gridsize;
  private CellType[][] _grid;


  public GameGrid(Pair gridsize)
  {
    _gridsize = gridsize;
    _grid = new CellType[_gridsize.x][_gridsize.y];
  }


  // IGridable

  public int get_width()
  {
    return _gridsize.y;
  }

  public int get_height()
  {
    return _gridsize.x;
  }

  public CellType node_value(int x, int y)
  {
    if (in_range(x, y))
      return _grid[x][y];
    else
      throw new RuntimeException("x or y out of range");
  }

  public boolean in_range(int x, int y)
  {
    return x >= 0 && x < get_height() && y >= 0 && y < get_width();
  }

  public boolean out_of_range(int x, int y)
  {
    return x < 0 || x >= get_height() || y < 0 || y >= get_width();
  }

  protected void set_node_value(int x, int y, CellType cell_value)
  {
    if (in_range(x, y))
      _grid[x][y] = cell_value;
    else
      throw new RuntimeException("x or y out of range");
  }
}